### PR TITLE
#380 Add net60 as additional target framework

### LINKED ISSUE #380: Add additional TFM to eliminate explicit dependency
**Is your feature request related to a problem? Please describe.**
I want to minimise dependencies in my project by utilising framework dependencies wherever possible

**Describe the solution you'd like**
I want the package to have an additional TFM (net 6) added so that the explicit dependency on System.Text.Json can be removed. 

**Describe alternatives you've considered**
Accept the additional dependency

**Additional context**
n/a

### PR DESCRIPTION
**Pull request by thompson-tomo**
Makes eliminated packages where they can be provided by the framework by using conditions and added net 6 to STJ so that it can also become optional.

Closes #380

### CODE REVIEW

**Comment by axunonb** [comment]
@thompson-tomo Thanks! Please have a look at the failing build on appveyor for this PR.

**Comment by thompson-tomo** [comment]
@axunonb have gone and added an additional dependency to ZString to resolve the build issue.

**Comment by thompson-tomo** [comment]
Hmm, there is 2 trains of thought:
- Keep framework's consistent across libraries
- only target needed frameworks ie based on needs (dependency reduction, AOT, trimming etc)

I don't have a strong opinion either way but do tend for send one due to package size reduction.

**Comment by axunonb** [comment]
Adding NET60 as a target framework to only 1-2 projects is not enough. We're ready to go this step, but for all SmartFormat projects. Cysharp/ZString released v2.6.0 lately, which also targets NET60 and NET70. All this requires a deeper look into all consequences and possible issues. Maybe we can finally reference Cysharp/ZString as a package directly.
So a first step to go might be to migrate Cysharp/ZString to v2.6.0, and afterwards add NET60 to all projects - including the conditional dependencies you're suggesting.

**Comment by thompson-tomo** [comment]
Have gone and cleaned up properties which has allowed me to bump all packages in 1 hit. I have overridden the TFM for the base package due to nullability issues which someone should look at.

**Comment by thompson-tomo** [comment]
Have gone and cleaned up properties which has allowed me to bump all packages in 1 hit. I have overriden the TFM for the base package due to nul

**Comment by axunonb** [comment]
We've created a [new branch v3.5.0](https://github.com/axuno/SmartFormat/tree/version/3.5.0).
It already includes necessary changes to reference the Cysharp `ZString` Nuget package.
To continue with this PR please rebase to this new branch. It would be great to cover all (extension) projects the way you propose for `System.Text.Json`.
After merging the PR, a final step is to update conditional compilation in the current code base, and to update package references to the latest versions. We'll start with this workload now.

**Comment by thompson-tomo** [comment]
@axunonb requested changes have been made and all builds now.

File: src/Demo/Demo.csproj [hunk 1]
```diff
@@ -2,10 +2,12 @@
 
   <PropertyGroup>
     <OutputType>WinExe</OutputType>
-    <TargetFramework>net6.0-windows</TargetFramework>
+    <TargetFrameworks>net6.0-windows</TargetFrameworks>
     <UseWindowsForms>true</UseWindowsForms>
     <ApplicationIcon>SmartFormat.Demo.ico</ApplicationIcon>
     <LangVersion>8.0</LangVersion>
+    <Nullable>disable</Nullable>
+    <NoWarn>1591,1573</NoWarn>
```
Comment (axunonb) [inline src/Demo/Demo.csproj:10]: Override Directory.Build.props

File: src/Performance_v27/Performance_v27.csproj [hunk 2]
```diff
@@ -9,6 +9,8 @@
     <SignAssembly>true</SignAssembly>
     <AssemblyName>Performance_v27</AssemblyName>
     <RootNamespace>Performance</RootNamespace>
+	<Nullable>disable</Nullable>
+    <NoWarn>1591,1573</NoWarn>
```
Comment (axunonb) [inline src/Performance_v27/Performance_v27.csproj:13]: Override Directory.Build.props

File: src/Performance/Performance.csproj [hunk 3]
```diff
@@ -2,10 +2,12 @@
 
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFramework>net6.0</TargetFramework>
+    <TargetFrameworks>net6.0</TargetFrameworks>
     <LangVersion>9.0</LangVersion>
     <AssemblyOriginatorKeyFile>../SmartFormat/SmartFormat.snk</AssemblyOriginatorKeyFile>
     <SignAssembly>true</SignAssembly>
+    <Nullable>disable</Nullable>
+	<NoWarn>1591,1573</NoWarn>
```
Comment (axunonb) [inline src/Performance/Performance.csproj:10]: Override Directory.Build.props

**Comment by axunonb** [comment]
LGTM, thanks for the PR and your initiative.

**Pull request merged**